src/app/containers/PageHandlers/withVariant/index.jsx

```diff
@@ -1,36 +1,63 @@
 import React from 'react';
+import pathToRegexp from 'path-to-regexp';
+import { pathOr } from 'ramda';
 import { useParams, useLocation, Redirect } from 'react-router-dom';
 import { variantSanitiser, getVariant } from '#lib/utilities/variantHandler';
 import {
   getPreferredVariantCookie,
   setPreferredVariantCookie,
 } from '#contexts/UserContext/cookies';
 
-const redirect = (location, variant) => {
+const redirect = (location, pathname) => {
   return (
     <Redirect
       to={{
         ...location,
-        pathname: `${location.pathname}/${variant}`,
+        pathname,
       }}
     />
   );
 };
 
+const getRedirectUrl = (path, params, variant) => {
+  const pathTo = pathToRegexp.compile(path);
+
+  return pathTo(
+    {
+      ...params,
+      variant: `/${variant}`,
+    },
+    {
+      encode: value => value,
+    },
+  );
+};
+
 const withVariant = Component => props => {
   const { service, variant } = useParams();
   const location = useLocation();
   const defaultVariant = getVariant({ service, variant });
   const sanitizedVariant = variantSanitiser(variant);
   const cookieVariant = getPreferredVariantCookie(service);
   const isNotDefaultVariant = defaultVariant !== 'default';
+  const { path } = pathOr({}, ['match'], props);
+
+  const getRedirectUrlFromVariant = variantOverride =>
+    getRedirectUrl(
+      path,
+      {
+        service,
+        variant,
+      },
+      variantOverride,
+    );
 
-  if (!sanitizedVariant && cookieVariant) {
```

Comment:
This turned out to be the culprit. `sanitizedVariant` is truthy when this is supposed to happen.